src/server/status_server.rs

```diff
@@ -234,11 +247,92 @@ impl StatusServer {
     }
 }
 
+// For handling fail points related requests
+#[cfg(not(feature = "no-fail"))]
+fn handle_fail_points_request(
+    req: Request<Body>,
+) -> Box<dyn Future<Item = Response<Body>, Error = hyper::Error> + Send> {
+    let path = req.uri().path().to_owned();
+    let method = req.method().to_owned();
+    let fail_path = format!("{}/", FAIL_POINTS_REQUEST_PATH);
```

Comment:
What if a user sends a request for `/fail_sufix`? It seems this function treat it like `/fail`.

Reply:
Yes. Not sure whether that would be a problem for now. Maybe we can try using actix-web to simplify all this a bit?

Reply:
I think you can verify if it equals to `FAIL_POINTS_REQUEST_PATH` when `fail_path_has_sub_path` is false.

Reply:
@BusyJay Thanks it is a good idea. I made another commit to check it and return 404 if the path is something like `/fail_sufix`